packages/atomic/src/components/atomic-search-box/atomic-search-box.tsx

```diff
@@ -41,21 +46,33 @@ export class AtomicSearchBox {
   private inputRef!: HTMLInputElement;
   private listRef!: HTMLElement;
   private querySetActions!: QuerySetActionCreators;
+  private pendingSuggestionEvents: SearchBoxSuggestionsEvent[] = [];
+  private suggestions: SearchBoxSuggestions[] = [];
 
   @BindStateToController('searchBox')
   @State()
   private searchBoxState!: SearchBoxState;
   @State() public error!: Error;
   @State() private isExpanded = false;
   @State() private activeDescendant = '';
+  @State() private suggestionElements: SearchBoxSuggestionElement[] = [];
+
+  /**
+   * The amount of queries displayed when the user interacts with the search box.
+   * By default, a mix of query suggestions and recent queries will be shown.
+   * You can configure those settings using the following components as children:
+   *  - atomic-search-box-query-suggestions
+   *  - atomic-search-box-recent-queries
+   */
+  @Prop() public numberOfQueries = 8;
```

Comment:
Like described in the documents, there will also be "numberOfResults" for the other types of suggestions. So these scenarios can be configured:

![Screen Shot 2021-09-28 at 3 26 15 PM](https://user-images.githubusercontent.com/4923043/135152803-e0bb8ba0-b0bd-4000-aea3-6527b11c23db.png)

Reply:
I think that instead of making this an option here, we could have an opiniated default, and if people want to override it then they need to specify `atomic-search-box-query-suggestions` and `atomic-search-box-recent-queries` themselves.

Reply:
This is why each of the sub elements, atomic-search-box-query-suggestions and atomic-search-box-recent-queries, have maximums, which when not defined, fill up the `numberOfQueries` value. How else could we configure the following:

- I want a max of 8 queries at all times (any type)
- When there is no query in the input, I want all the recent queries I can up to 8, but fill the rest with query suggestions
- When there is a query, I want up to 3 matching recent queries, and fill the rest with query suggestions